android/app/src/main/java/org/fossasia/openevent/viewmodels/SponsorsFragmentViewModel.java

```diff
@@ -0,0 +1,36 @@
+package org.fossasia.openevent.viewmodels;
+
+import android.arch.lifecycle.LiveData;
+import android.arch.lifecycle.MutableLiveData;
+import android.arch.lifecycle.Transformations;
+import android.arch.lifecycle.ViewModel;
+
+import org.fossasia.openevent.data.Sponsor;
+import org.fossasia.openevent.dbutils.LiveRealmData;
+import org.fossasia.openevent.dbutils.RealmDataRepository;
+
+import java.util.List;
+
+public class SponsorsFragmentViewModel extends ViewModel {
+
+    private LiveData<List<Sponsor>> sponsorsList;
+    private RealmDataRepository realmRepo;
+
+    public SponsorsFragmentViewModel() {
+        realmRepo = RealmDataRepository.getDefaultInstance();
+    }
+
+    public LiveData<List<Sponsor>> getSponsors() {
+        if (sponsorsList == null) {
+            sponsorsList = new MutableLiveData<>();
```

Comment:
That's exactly what shouldn't happen. Assignment should happen once and only once

Reply:
Okay. I hope I did it correctly this time